week2-1/main.js

```diff
@@ -0,0 +1,108 @@
+import pic1 from "./assets/김규민.jpeg";
+import pic2 from "./assets/전희선.jpeg";
+import pic3 from "./assets/서혜은.jpg";
+import pic4 from "./assets/황주희.jpeg";
+import pic5 from "./assets/백지연.png";
+
+const $ = (selector) => document.querySelector(selector);
+
+let currentStep;
+
+const quizList = [
+  {
+    src: pic1,
+    answer: "김규민",
+  },
+  {
+    src: pic2,
+    answer: "전희선",
+  },
+  {
+    src: pic3,
+    answer: "서혜은",
+  },
+  {
+    src: pic4,
+    answer: "황주희",
+  },
+  {
+    src: pic5,
+    answer: "백지연",
+  },
+];
+
+function initGame({score, image}){
+  currentStep = 0;
+  score.innerText = 0;
+  image.src = quizList[currentStep].src;
+}
+
+function showModal(modalContent, keepOpen){
+  const modal = $('.modal');
+  const modalBody = $('p.modal__body');
+  //심화과제 2번 구현
+  modal.addEventListener('click', () => {modal.classList.add('hide')}); 
+  modalBody.innerHTML = modalContent;
+  modal.classList.remove('hide');
+  if (keepOpen) return;
+  setTimeout(()=>{
+    modal.classList.add('hide');
+  }, 500);
+}
+// 심화과제 1번 구현
+function showLoadingModal(image){
+    image.onload = () => {
+      showModal(`
+        로딩이라귯     
+      `)
+    }
+}
+
+function goNextStep(score, image){
+  $('.scoreBoard').classList.remove('scored');
+  currentStep++;
+  score.innerText = +score.innerText + 1;
+  $('.scoreBoard').classList.add('scored'); // 심화과제 3번 구현
+  if(currentStep === quizList.length){
+    showModal(`
+    <a href="/">메인화면으로</a>
+  `, true)
+  } else {
+    showLoadingModal(image);
+    image.src = quizList[currentStep].src;
+  }
+}
+
+function attachEvent({score, image, answer, shuffle}){
+  shuffle.addEventListener('click', ()=> {// 다시하기 버튼 기능 구현
+    gameManager({score, image});
+  });
```

Comment:
```suggestion
  shuffle.addEventListener('click', ()=> {// 다시하기 버튼 기능 구현
    initGame({ score, image });
  });
```

`gameManager`에는 `initGame`이랑 `attachEvent` 둘 다 들어있는데,
`initGame`만 필요하니까 이렇게 수정하는 게 좋을 것 같습니다~!
그리고 이렇게 수정하면 다시하기 버튼 눌렀을 때 뜨는 에러 해결되네요 👍